src/InvariantTheory/invariant_rings.jl

```diff
@@ -63,10 +63,12 @@ function invariant_ring(R::MPolyDecRing, M::Vector{<: MatrixElem})
   return invariant_ring(R, matrix_group([change_base_ring(K, g) for g in M]))
 end
 
-invariant_ring(matrices::MatrixElem{T}...) where {T} = invariant_ring(collect(matrices))
+function invariant_ring(m::MatrixElem{T}, ms::MatrixElem{T}...) where {T} 
+  return invariant_ring([m, ms...])
+end
 
-function invariant_ring(R::MPolyDecRing, matrices::MatrixElem{T}...) where {T}
-  return invariant_ring(R, collect(matrices))
+function invariant_ring(R::MPolyDecRing, m::MatrixElem{T}, ms::MatrixElem{T}...) where {T} 
+  return invariant_ring(R, [m, ms...])
```

Comment:
Dumb question: is this faster/better/prettier than what was there before?
Looking at this signature, I would now expect that `m` and `ms` have different meanings in this function and not that they are just put together in the same vector.
(If it is only about the `T` we could do
```suggestion
function invariant_ring(R::MPolyDecRing, matrices::MatrixElem...)
  return invariant_ring(R, collect(matrices))
```
as the `T` is not used... If somebody puts in matrices of different types, this will error if the entries cannot be cast in the coefficient ring of the polynomial ring.)

Reply:
This is about the existence of a dispatch where the T is not instantiated.

Your suggestion would be (mostly) fine as well. The single drawback is, that the type of `collect(matrices)` gets inferred as `Vector{Any}` for the 0-arg case.

Reply:
(I think we should actually throw a meaningful error if there are no matrices, but that's off-topic for now.)
I can't say I can see the ramification of this function not having a completely well-defined return type. I find the new signature confusing (for the human); if it is less confusing for the computer, that's fine by me.